lisa/tests/kernel/scheduler/load_tracking.py

```diff
@@ -733,3 +735,218 @@ def test_load_avg_behaviour(self, error_margin_pct=5, allowed_error_pct=5)\
+
+        phase_start = sw_df[sw_df.next_comm == list(self.rtapp_profile.keys())[0]].index[0]
+
+        for phase in range(self.nr_phases):
+            # Start looking at signals once they should've converged
+            start = phase_start + UTIL_AVG_CONVERGENCE_TIME_S
+            # Trim the end a bit, otherwise we could have one or two events
+            # from the next phase
+            end = phase_start + self.phases_durations[phase] * .9
+            phase_df = df[start:end]
+
+            for cpu in self.cpus:
+                util = phase_df[phase_df.cpu == cpu].util
+                cpu_util[cpu][phase] = area_under_curve(util) / (end - start)
+
+            phase_start += self.phases_durations[phase]
+
+        return cpu_util
+
+    def test_util_task_migration(self, allowed_error_pct=5) -> ResultBundle:
+        """
+        Test that a migrated task properly propagates its utilization at the CPU level
+
+        :param allowed_error_pct: How much the trace averages can stray from the
+          expected values
+        :type allowed_error_pct: float
+        """
+        expected_cpu_util = self.get_expected_cpu_util()
+        trace_cpu_util = self.get_trace_cpu_util()
+
+        passed = True
+
+        expected_metrics = {}
+        trace_metrics = {}
+        deltas = {}
+
+        for cpu in self.cpus:
+            cpu_str = "cpu{}".format(cpu)
+
+            expected_metrics[cpu_str] = TestMetric({})
+            trace_metrics[cpu_str] = TestMetric({})
+            deltas[cpu_str] = TestMetric({})
+
+            for phase in range(self.nr_phases):
+                if not self.is_almost_equal(
+                        trace_cpu_util[cpu][phase],
+                        expected_cpu_util[cpu][phase],
+                        allowed_error_pct):
+                    passed = False
+
+                # Just some verbose metric collection...
+                phase_str = "phase{}".format(phase)
+
+                expected_metrics[cpu_str].data[phase_str] = TestMetric(
+                    expected_cpu_util[cpu][phase])
+                trace_metrics[cpu_str].data[phase_str] = TestMetric(
+                    trace_cpu_util[cpu][phase])
+                deltas[cpu_str].data[phase_str] = TestMetric(
+                    -100 * (expected_cpu_util[cpu][phase] - trace_cpu_util[cpu][phase]) /
+                    expected_cpu_util[cpu][phase], "%")
+
+        res = ResultBundle.from_bool(passed)
+        res.add_metric("Expected utilization", expected_metrics)
+        res.add_metric("Trace utilization", trace_metrics)
+        res.add_metric("Utilization deltas", deltas)
+
+        return res
+
+class OneTaskCPUMigration(CPUMigrationBase):
+    """
+    Some tasks on two big CPUs, one of them migrates in its second phase.
+    """
+
+    @classmethod
+    def get_migration_cpus(cls, te):
+        cpu_capacities = te.plat_info["cpu-capacities"]
+
+        return [cpu for cpu, capa in cpu_capacities.items()
```

Comment:
why not just "cap" ? :)

Reply:
Because otherwise my mind [gets sidetracked](https://www.youtube.com/watch?v=1dWIPUuTPB4) whenever I read this line.

Reply:
Fine... although I would have pick a different name (e.g. `cpu_cap`)...